### ISSUE TITLE
Q) onert ExecutionBuilder releaseCodeMap() method use std::move ?

### ISSUE THREAD

**Issue by seanshpark** [Samsung/ONE#5075]
I'm curious why return with `std::move()` for `releaseCodeMap()`?
Could anyone good at C++ answer this?

https://github.com/Samsung/ONE/blob/a892ba0fbb17446006057cb389086e63e27a82b6/runtime/onert/core/include/compiler/ExecutionBuilder.h#L40

where `CodeMap` is https://github.com/Samsung/ONE/blob/a892ba0fbb17446006057cb389086e63e27a82b6/runtime/onert/core/include/compiler/CodeMap.h#L40

Although there is `unique_ptr` in `CodeAndInfo`
https://github.com/Samsung/ONE/blob/a892ba0fbb17446006057cb389086e63e27a82b6/runtime/onert/core/include/compiler/CodeMap.h#L27-L32

**Comment by wateret** [comment]
As `ExecutionBuilder` generates the results in multiple steps(`append` is called repeatedly, it is not a single function like `CodeMap build()`), so it contains the result(`_code_map`) as a member. And after finishing execution building, user needs to get the result, but it has `unique_ptr` so we can't easily copy it. So I chose to move it. It is also good for performance(move is faster than copy).

FYI, `op_seq` and `lower_info` are pointers so it is just same as assignment. `fn_seq` is `unique_ptr` so it actually moves the object. So after calling `releaseCodeMap`, `_code_map` has been moved, so the user will get a dummy object for the next calls.
